client/src/components/Tool/ToolForm.vue

```diff
@@ -0,0 +1,300 @@
+                        value="false"
+                        @onChange="onChangeReuse"
+                    />
+                </template>
+                <template v-slot:buttons>
+                    <ButtonSpinner id="execute" title="Execute" @onClick="onExecute" :wait="showExecuting" />
+                </template>
+            </ToolCard>
+        </div>
+    </CurrentUser>
+</template>
+
+<script>
+import Scroller from "vue-scrollto";
+import { getAppRoot } from "onload/loadConfig";
+import { getGalaxyInstance } from "app";
+import { getTool, submitJob, updateTool } from "./services";
+import { send } from "./utilities";
+import ToolCard from "./ToolCard";
+import ButtonSpinner from "components/Common/ButtonSpinner";
+import CurrentUser from "components/providers/CurrentUser";
+import LoadingSpan from "components/LoadingSpan";
+import Form from "components/Form/Form";
+import FormElement from "components/Form/FormElement";
+import ToolSuccess from "./ToolSuccess";
+import Webhook from "components/Common/Webhook";
+
+export default {
+    components: {
+        ButtonSpinner,
+        CurrentUser,
+        LoadingSpan,
+        Form,
+        ToolCard,
+        FormElement,
+        ToolSuccess,
+        Webhook,
+    },
+    props: {
+        id: {
+            type: String,
+            default: null,
+        },
+        version: {
+            type: String,
+            default: null,
+        },
+        job_id: {
+            type: String,
+            default: null,
+        },
+    },
+    data() {
+        return {
+            showLoading: true,
+            showForm: false,
+            showEntryPoints: false,
+            showRecommendation: false,
+            showSuccess: false,
+            showError: false,
+            showExecuting: false,
+            error: null,
+            formConfig: {},
+            formData: {},
+            errorTitle: "",
+            errorContent: null,
+            messageVariant: "",
+            messageText: "",
+            reuse: false,
+            email: false,
+            remap: false,
+            entryPoints: [],
+            jobDef: {},
+            jobResponse: {},
+            validationInternal: null,
+            validationScrollTo: null,
+            validationErrors: null,
+            currentVersion: this.version,
+        };
+    },
+    created() {
+        this.requestTool();
+        const Galaxy = getGalaxyInstance();
+        if (Galaxy.currHistoryPanel) {
+            Galaxy.currHistoryPanel.collection.on("change", () => {
+                this.onUpdate();
+            });
+        }
+    },
+    computed: {
+        toolName() {
+            return this.formConfig.name;
+        },
+        errorContentPretty() {
+            return JSON.stringify(this.errorContent, null, 4);
+        },
+        emailAllowed() {
+            const Galaxy = getGalaxyInstance();
+            return Galaxy.config.server_mail_configured && !Galaxy.user.isAnonymous();
```

Comment:
Yes. Either separate out the CurrentUser and Config providers or use the corresponding vuex store methods directly. I think separation is better. Instead of one big component this should be one big component and one small component whose job is to assemble dependencies like this:

```html
<!-- instead of --->
<ToolForm v-bind="yourProps" />

<!-- New ToolForm.vue, make it 2 layers -->
<ConfigProvider v-slot="{ config }">
<CurrentUser v-slot="{ user }">
     <!-- this is basically the existing ToolForm with all the dependencies removed and turned into props
     You can transparently bind events and properties with the magic keywords ($attrs and $listeners) as I've shown below. 
     To the outside, this will look exactly like TOolForm looks now with all the props and listeners passing through  -->
     <PreviousToolForm :config="config" :user="user" v-bind="$attrs" v-on="$listeners" />
</CurrentUser>
</ConfigProvider>
```

Reply:
Added config and user providers.